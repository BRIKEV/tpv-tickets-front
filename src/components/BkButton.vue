<template>
  <button
    class="btn"
    @click="btnClicked">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'BkButton',

  methods: {
    btnClicked() {
      this.$emit('btn-clicked');
    },
  },

};
</script>

<style lang="scss" scoped>
  .btn {
    background: #1a2446;
    border-radius: 8px;
    color: #fff;
    height: 40px;
    text-align: center;
    text-decoration: none;
    font-weight: bold;
    height: 40px;
    font-size: 0.9375rem;
    transition: all .15s linear;
    border: 1px solid #FFF;
    &:hover {
      background: #118060;
    }

    &:active {
      background-color: #0d5f48;
    }
  }
</style>
