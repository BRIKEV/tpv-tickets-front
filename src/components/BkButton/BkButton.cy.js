import { mount } from "@cypress/vue"
import Bkbutton from "./BkButton.vue"

describe('<Bkbutton />', () => {
  it('renders', () => {
    // see: https://vue-test-utils.vuejs.org/
    mount(Bkbutton, { props: {} });
  });
});
