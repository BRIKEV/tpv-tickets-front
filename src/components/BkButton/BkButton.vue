<template>
  <button
    class="btn"
    :disabled="disabled"
    @click="btnClicked">
    <beat-loader :loading="isLoading" :color="'white'" :size="10" />
    <p class ="content" v-if="!isLoading">
      <slot></slot>
    </p>
  </button>
</template>

<script>
export default {
  name: 'BkButton',

  props: {
    isLoading: {
      type: Boolean,
      default: false,
    },
    disabled: Boolean,
  },

  methods: {
    btnClicked() {
      this.$emit('btn-clicked');
    },
  },

};
</script>

<style lang="scss" scoped>
  @import "@/theme/index.scss";

  .btn {
    background: $brand;
    border-radius: 8px;
    color: $white;
    height: 45px;
    text-align: center;
    text-decoration: none;
    font-weight: bold;
    height: 45px;
    font-size: $fs-base;
    transition: all .15s linear;
    border: 1px solid $white;
    &:hover {
      background: lighten($brand, 20%);
    }

    &:active {
      background-color: $brand;
    }

    &:disabled {
      background: lightgrey;
    }

    .content {
      margin: 0;
    }
  }
</style>
