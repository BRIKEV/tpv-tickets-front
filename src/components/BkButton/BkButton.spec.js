import { mount } from '@vue/test-utils';
import BkButton from '@/components/BkButton/BkButton.vue';

describe('BkButton component', () => {
  it('Should be render correctly', () => {
    const wrapper = mount(BkButton);
    expect(wrapper.element).toMatchSnapshot();
  });
});
