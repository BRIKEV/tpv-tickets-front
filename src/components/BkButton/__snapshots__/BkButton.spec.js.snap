// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BkButton component Should be render correctly 1`] = `
<button
  class="btn"
>
  <beat-loader
    color="white"
    size="10"
  />
   
  <p
    class="content"
  />
</button>
`;
